sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/WindowMappingFn.java

```diff
@@ -19,15 +19,20 @@
 package org.apache.beam.sdk.transforms.windowing;
 
 import java.io.Serializable;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Internal;
 import org.apache.beam.sdk.transforms.ParDo.MultiOutput;
 import org.apache.beam.sdk.values.PCollectionView;
 import org.joda.time.Duration;
 
 /**
+ * <b>For internal use only; no backwards compatibility guarantees.</b>
+ *
  * A function that takes the windows of elements in a main input and maps them to the appropriate
  * window in a {@link PCollectionView} consumed as a
  * {@link MultiOutput#withSideInputs(PCollectionView[]) side input}.
  */
+@Internal
```

Comment:
Done.

We do expect users to use it eventually, but probably a long time from now and very rarely.